File: digitcounts/digitcounts.py
class Solution:
    # @param k & n  two integer
    # @return ans a integer
    def digitCounts(self, k, n):
        digit = 0
        for i in range(n+1):
            j = i
            while True:
                if j%10 == k:
                    digit+=1
                j /= 10
                if j == 0:
                    break
        return digit

if __name__ == "__main__":
    test = Solution()
    print(test.digitCounts(0,12))

File: digitcounts/digitcounts.java
class Solution {
    /*
     * param k : As description.
     * param n : As description.
     * return: An integer denote the count of digit k in 1..n
     */
    public int digitCounts(int k, int n) {
        int digit = 0;
        for(int i = 0; i <= n; i++){
	    	int j = i;
	    	do{
	    		if(j % 10 == k)
	    		{
	    			digit++;
	    		}
	    		j /= 10;
	    	}while(j > 0);
		}
		return digit++;
    }
};

public class digitcounts{
    public static void main(String argv[])
    {
        Solution test = new Solution();
        System.out.println(test.digitCounts(0,12));
    }
}
